File: Server/src/model/Bullet/BulletFactory.java
package model.Bullet;

import model.Tower.AttackTower;
import model.Tower.Tower;

public class BulletFactory {
    public static Bullet create(AttackTower tower){
        if (tower.getArchetype().equals("magic")){
            return new MagicBullet(tower);
        }
        if (tower.getBulletType().equals("chasing")){
            return new ChasingBullet(tower);
        }

        if (tower.getBulletType().equals("boomerang")){
            return new BoomerangBullet(tower);
        }
        if (tower.getBulletType().equals("straight") ){
            return new PositionBullet(tower);
        }

        return null;
    }
}

File: Client/src/modules/models/bullet/BulletFactory.js
var BulletFactory = cc.Class.extend({
});
BulletFactory.create = function(tower){
    if (tower.archetype == GC.TOWER.TYPES.MAGIC){
        return new MagicBullet(tower);
    }
    if (tower.bulletType == GC.TOWER.BULLET_TYPES.CHASING){
        return new ChasingBullet(tower);
    }
    if (tower.bulletType == GC.TOWER.BULLET_TYPES.BOOMERANG){
        return new BoomerangBullet(tower);
    }
    if (tower.bulletType == GC.TOWER.BULLET_TYPES.POSITION){
        return new PositionBullet(tower);
    }
    return null;
}